src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -81,20 +81,18 @@ public void Dispose()
         public bool MoveNext(CancellationToken cancellationToken = default(CancellationToken))
         {
             bool hasMore;
-            try
+            while (true)
```

Comment:
This isn't truly an infinite loop. We will loop until either MoveNext returns without an error or Resume fails.